nntrainer/src/databuffer.cpp

```diff
@@ -374,10 +383,15 @@ int DataBuffer::init() {
   this->cur_val_bufsize = 0;
   this->cur_test_bufsize = 0;
 
+  readyTrainData.lock();
```

Comment:
do we need locking here? this is init and no thread is active yet.

Reply:
coverity says we need.

Reply:
ok